classes/mail/variables/SubmissionEmailVariable.php

```diff
@@ -35,13 +39,15 @@ class SubmissionEmailVariable extends Variable
     public const SUBMISSION_URL = 'submissionUrl';
 
     protected Submission $submission;
-
     protected Publication $currentPublication;
+    protected Context $context;
 
     public function __construct(Submission $submission)
     {
         $this->submission = $submission;
         $this->currentPublication = $this->submission->getCurrentPublication();
+        $contextDao = Application::getContextDAO();
+        $this->context = $contextDao->getById($this->submission->getData('contextId'));
```

Comment:
We don't want to load the context from the database with every email variable, because it's an expensive call from a performance perspective. The same problem exists in the `ReviewAssignmentEmailVariable` and maybe others.

Can't we re-use the context from the `Mailable`? Maybe by passing the `Mailable` into each `Variable` class when it is constructed?

Reply:
Yes, database calls are used to retrieve some of the template variables. I thought about that by didn't really see any performance issues during testing, e.g., when sending bulk emails.

Reply:
The performance issue it causes will depend on a lot of factors, including how large the context settings are, how many contexts there, and how many times it must be pulled from the database. We shouldn't go back to the database for something we already have in memory.

Reply:
> Can't we re-use the context from the Mailable? Maybe by passing the Mailable into each Variable class when it is constructed?

It's not only about the context, e.g., I retrieve `StageAssignment` from the DAO in `SubmissionEmailVariable`. Basically, each `Variable` object should be aware of all variables rather than only associated. Are you OK with this approach?

Reply:
It looks like that is used only for `submittingAuthorName`, which is only used in one `Mailable`. That should probably be an extension of that `Mailable` class, rather than a general variable every time a submission is involved.

> Basically, each Variable object should be aware of all variables rather than only associated.

The dependencies of one email variable on another are likely to get complicated. But let's try to work around such cases as one-offs for now. In the future we may need to give all variables access to the other variables -- or at least what's in the mailable. But hopefully we can defer that to after 3.4.